.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: Exaile CI
+
+on: [push, pull_request]
+
+jobs:
+
+  lint:
+    timeout-minutes: 1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: pip install black
+      - name: Check format
+        run: make check_format
+
+  test:
+    strategy:
+      matrix:
+        imgtag:
+          - debian10-python3
+          - fedora31-python3
+          - ubuntu18.04-python3
+          - ubuntu19.10-python3
+    timeout-minutes: 4
+    runs-on: ubuntu-latest
+    container: exaile/exaile-testimg:${{ matrix.imgtag }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build and run tests
+        # The container always runs as root regardless of how we set it up in
+        # the Dockerfile. Exaile has tests for file permissions that fail in
+        # that setup, so we de-escalate manually here.
+        run: |
+          useradd -MN exaile
+          chown -R exaile .
+          export HOME=/tmp/home
+          su -m exaile -c "make BUILDDIR=/tmp/build test test_compile check-doc"
+
+  deploy:
+    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
I've been using `if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')`... I don't remember if the push actually matters, I copied it from elsewhere.

Reply:
It should be ok in this case (this workflow only listens to `push` and `pull_request` events, and the latter [doesn't contain `ref`](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#pull_request)), but I'll add it anyway just to be explicit.